TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/HttpJsonMessageWithFaultingPayload.java

```diff
@@ -48,11 +49,24 @@ public void setProtocol(String value) {
     }
 
     @Override
-    public Map<String, Object> headersMap() {
-        return Collections.unmodifiableMap(headers());
+    public Map<String, List<String>> headers() {
```

Comment:
This function is a great example of why you want to move this to a separate adapter class.  It looks just like a plain old getter, but it does a significant amount of construction.  Imagine somebody calling this from the inside of a loop.  Given that you only need _this_ interface for the adapter, you should move it out so that this method isn't confused to be an intrinsically, first-class, 'native' function on the class.

Reply:
Addressed, thanks for the comment